storage_service/locations/models/dspace.py

```diff
@@ -121,6 +128,58 @@ def _get_metadata(self, input_path, aip_uuid):
         os.remove(mets_path)
         return kwargs
 
+    def _archive(self, src, dst):
+        """
+        Run the archiver in order to create the archive file based on the
+        preferred format. `dst` is the path of the archive file. The file
+        extension must not be included, instead this function will return
+        the final destination with the extension on it according to the
+        archive format preferred.
+        """
+        if self.archive_format == self.ARCHIVE_FORMAT_ZIP:
+            dst, command = self._archive_zip(src, dst)
+        elif self.archive_format == self.ARCHIVE_FORMAT_7Z:
+            dst, command = self._archive_7z(src, dst)
+        else:
+            raise ValueError('Archive format not supported')
+
+        try:
+            subprocess.check_call(command)
+        except subprocess.CalledProcessError:
+            LOGGER.error('Could not compress %s', src)
+            raise
+
+        return dst
+
+    def _archive_zip(self, src, dst):
+        """ Return the command that creates the ZIP archive file. """
+        if not dst.endswith('.zip'):
+            dst += '.zip'
+
+        return (dst, [
+            'zip',
+            '-r',
+            dst,  # Destination (without extension)
```

Comment:
Should `dst` have the extension here? The comment says no but the current behaviour is yes.